openquake/mbt/tests/tools/fault_modeler/test_fault_modeling_utils.py

```diff
@@ -0,0 +1,294 @@
+
+        fault = {'coords': [[0., 0.], [0., 1.]],
+                 'upper_seis_depth': 0.,
+                 'lower_seis_depth': 10.,
+                 'average_dip': '(30,,)',
+                 'dip_dir': 'E'}
+
+        width = calc_fault_width_from_usd_lsd_dip(fault,
+                                                  param_map=self.param_map,
+                                                  defaults=defaults)
+
+        self.assertTrue(abs(20. - width) < 0.01)
+
+    def test_get_fault_width_seismo_depth(self):
+
+        fault = {'coords': [[0., 0.], [0., 1.]],
+                 'upper_seis_depth': 0.,
+                 'lower_seis_depth': 10.,
+                 'average_dip': '(30,,)',
+                 'dip_dir': 'E'}
+
+        width = get_fault_width(fault, width_method='seismo_depth',
+                                param_map=self.param_map,
+                                defaults=defaults)
+
+        self.assertTrue(abs(20. - width) < 0.01)
+
+    def test_get_lsd_from_width_scaling_rel(self):
+        """
+        Tests lower seismogenic depth (from length scaling relationship)
+        by asserting the seismogenic thickness should be half the fault
+        width for a fault with 30 degree dip
+        """
+
+        fault = {'coords': [[0., 0.], [0., 1.]],
+                 'upper_seis_depth': 0.,
+                 'lower_seis_depth': 10.,
+                 'average_dip': '(30,,)',
+                 'dip_dir': 'E',
+                 'slip_type': 'Reverse'}
+
+        width = get_fault_width(fault, width_method='length_scaling',
+                                defaults=defaults, param_map=self.param_map)
+
+        lsd = get_lower_seismo_depth(fault, width_method='length_scaling',
+                                     defaults=defaults,
+                                     param_map=self.param_map)
+
+        seis_thickness = lsd - fault['upper_seis_depth']
+
+        self.assertAlmostEqual(seis_thickness * 2, width)
+
+    def test_get_fault_area_simple(self):
+
+        fault = {'coords': [[0., 0.], [0., 1.]],
+                 'upper_seis_depth': 0.,
+                 'lower_seis_depth': 10.,
+                 'average_dip': '(30,,)',
+                 'dip_dir': 'E',
+                 'slip_type': 'Reverse'}
+
+        length = get_fault_length(fault, defaults=defaults,
+                                  param_map=self.param_map)
+
+        area = get_fault_area(fault, area_method='simple',
+                              width_method='seismo_depth',
+                              param_map=self.param_map, defaults=defaults)
+
+        self.assertTrue(abs((length * 20.) - area) < 0.01)
+
+    # Rates
+    def test_get_net_slip_rate(self):
+
+        nsr = get_net_slip_rate(self.fault_1, slip_class='mle',
+                                param_map=self.param_map,
+                                defaults=defaults)
+
+        self.assertEqual(nsr, 6.)
+
+    def test_net_slip_from_strike_slip_fault_geom(self):
```

Comment:
Maybe a
```python
@unittest.skip("to be implemented")
```
decorator could be a better reminder for all tests not yet implemented.

Reply:
fixed.